src/CodeGen_Hexagon.h

```diff
@@ -0,0 +1,132 @@
+#ifndef HALIDE_CODEGEN_HEXAGON_H
+#define HALIDE_CODEGEN_HEXAGON_H
+
+/** \file
+ * Defines the code-generator for producing Hexagon machine code
+ */
+
+#if WITH_HEXAGON
```

Comment:
Can't refer to WITH_HEXAGON or LLVM_VERSION in a header. This goes into Halide.h, and users wont have these set. It would normally be equivalent to both of them being unset, but I don't think that's what you want. I hit an error in some user code which set WITH_HEXAGON but not LLVM_VERSION